javaparser-core/src/main/java/com/github/javaparser/ast/expr/StringLiteralExpr.java

```diff
@@ -78,6 +78,10 @@ public boolean remove(Node node) {
         return super.remove(node);
     }
 
+    public String asString() {
+        return value;
+    }
```

Comment:
If we do that, `StringLiteralExpr(String)` would do something different than `StringLiteralExpr.setValue(String)` which is unexpected. We can't get `setValue` to escape strings since all setters simply set the field without extra's, don't want to make an exception there.

Okay, after a lot of thinking: let's do your thing.
- make the constructor escape only the values that will create an illegal string constant (so \n and \r as `StringLiteralExpr.escape` does? Try reusing that and marking it deprecated)
- but also add a `setEscapedValue` or `setValueSafe` or whatever you think is clearest that does what the constructor does.
- `StringLiteralExpr(Range, String)` should stay as it is now!